File: Semana 3.1 - Repaso Singleton y Faqctory Method/Taller/Vehiculos/getIdentidad.cs
using System;
using System.Text;

public class GetIdentidad
{
    private string nombre = "Jarvic Balanta";
    private string patron;
    private string encabezado;

    public GetIdentidad(string _patron, string encabezado)
    {
        this.patron = _patron;
        this.encabezado = encabezado;
    }

    private string GetBase64(string input)
    {
        byte[] bytes = Encoding.UTF8.GetBytes(input);
        return Convert.ToBase64String(bytes);
    }

    public void getPatron()
    {
        Console.WriteLine("---------------------------------------------------------");
        Console.WriteLine("Patrón Identificado: " + patron);
        Console.WriteLine("---------------------------------------------------------");
    }

    public void GetNombre()
    {
        Console.WriteLine("---------------------------------------------------------");
        Console.WriteLine("Proyecto realizado por: " + GetBase64(nombre));
        Console.WriteLine("---------------------------------------------------------");

    }

    public void GetEncabezado()
    {
        string fecha = DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss");
        Console.WriteLine("---------------------------------------------------------");
        Console.WriteLine(" Materia: Patrones de Diseño - UCC");
        Console.WriteLine(" Fecha de ejecución: " + fecha);
        Console.WriteLine(" Docente: Harold Adrian");
        //Console.WriteLine(" Patrón de diseño: " + patron);
        Console.WriteLine(" Encabezado: " + encabezado);
        Console.WriteLine("---------------------------------------------------------");
    }
}

File: Semana_12_Patrones_Comportamiento/State/GetIdentidad.java
package Semana_12_Patrones_Comportamiento.State;

import java.util.Base64;
import java.text.SimpleDateFormat;
import java.util.Date;

public class GetIdentidad {
    private String nombre = "Jarvic Balanta";
    private String patron;
    private String encabezado;

    public GetIdentidad(String patron, String encabezado) {
        this.patron = patron;
        this.encabezado = encabezado;
    }

    private String getBase64(String input) {
        return Base64.getEncoder().encodeToString(input.getBytes());
    }

    public void getPatron() {
        System.out.println("---------------------------------------------------------");
        System.out.println("Patrón Identificado: " + patron);
        System.out.println("---------------------------------------------------------");
    }

    public void getNombre() {
        System.out.println("---------------------------------------------------------");
        System.out.println("Proyecto realizado por: " + getBase64(nombre));
        System.out.println("---------------------------------------------------------");
    }

    public void getEncabezado() {
        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
        String fecha = sdf.format(new Date());
        System.out.println("---------------------------------------------------------");
        System.out.println("Materia: Patrones de Diseño - UCC");
        System.out.println("Fecha de ejecución: " + fecha);
        System.out.println("Docente: Harold Adrian");
        // System.out.println("Patrón de diseño: " + patron);
        System.out.println("Encabezado: " + encabezado);
        System.out.println("---------------------------------------------------------");
    }
}
